client/src/components/pages/Profile/index.js

```diff
@@ -12,13 +12,12 @@ class Profile extends Component {
   state = { userInfo: null };
 
   componentDidMount() {
-    const {
-      location: {
-        state: { userInfo },
-      },
-    } = this.props;
-
-    if (userInfo) {
+    if (this.props.location.state.userInfo) {
+      const {
+        location: {
+          state: { userInfo },
+        },
+      } = this.props;
```

Comment:
you're using `this.props.location.state.userInfo` in the `if` and then inside you destructure it into a variable - destructure it first at the beginning of the function.
Btw having 5 lines just to destructure a simple variable kind of defeats the purpose, you can simplify it to
```js
const { userInfo } = this.props.location.state;
```

Reply:
if the state `undefined` we will get an error in the destruction

Reply:
good point although in this case the location.state is required by the propTypes. However looking at the object, why is `userInfo` even wrapped in location.state? It doesn't serve any purpose! `userInfo` should be its own top level property

Reply:
because we do not want to make a request to the server 
and the `userInfo` with us so we need to pass it via `Link`

Reply:
right it's the state from the router, kinda forgot about that 🙂